nsis/src/nsis/format/NsisBlockHeader.java

```diff
@@ -11,46 +11,34 @@
 
 public class NsisBlockHeader implements StructConverter {
 	private int offset;
-	private int num;
+	private int nbEntries;
+	private final static Structure STRUCTURE;
 
-	public NsisBlockHeader(BinaryReader reader) {
-		setOffset(0);
-		setNum(0);
-
-		try {
-			setOffset(reader.readNextInt());
-			setNum(reader.readNextInt());
-		} catch (IOException e) {
-			e.printStackTrace();
-		}
+	static {
+		STRUCTURE = new StructureDataType("block_header", 0);
+		STRUCTURE.add(DWORD, DWORD.getLength(), "offset", "Offset at which the block header starts");
+		STRUCTURE.add(DWORD, DWORD.getLength(), "num", "Number of entries in the block header");
+	}
 
+	public NsisBlockHeader(BinaryReader reader) throws IOException {
+		this.offset = reader.readNextInt();
+		this.nbEntries = reader.readNextInt();
 	}
 
 	@Override
 	public DataType toDataType() throws DuplicateNameException, IOException {
-		Structure structure = new StructureDataType("block_header", 0);
-		structure.add(DWORD, 4, "offset", null);
-		structure.add(DWORD, 4, "nb_entries", null);
-		return structure;
+		return STRUCTURE;
 	}
 
 	public int getOffset() {
-		return offset;
-	}
-
-	public void setOffset(int offset) {
-		this.offset = offset;
-	}
-
-	public int getNum() {
-		return num;
+		return this.offset;
 	}
 
-	public void setNum(int num) {
-		this.num = num;
+	public int getNbEntries() {
```

Comment:
nit for a better name: getEntriesNumber or getEntriesNum. I subjectively feel "num" is more commonly used for "number" than "nb".